File: kurs programowania/lista3/zadanie1/Romb.java
public class Romb extends Czworokat {

    public Romb(int a, int alfa) {
        super(a, a, a, a, alfa);

        if (alfa > 180)
            throw new IllegalArgumentException();
    }

    public double pole() {

        return Math.abs(this.bok1 * this.bok1 * Math.sin(Math.toRadians(
                Math.abs(kat - 2 * (kat % 90)))));
    }

    public double obwod() {

        return 4 * this.bok1;
    }
}

File: kurs programowania/lista3/cpp/Romb.cpp
#include "Romb.hpp"

#include <iostream>
#include <math.h>

Romb::Romb(int a, int alpha) : Czworokat(a, a, a, a, alpha)
{
    if (alpha > 180)
        std::runtime_error("illegal_args");
}

double Romb::pole()
{
    double radian = (abs(alpha - 2 * (alpha % 90)) * (3.14159265359 / 180));

    return a * a * sin(radian);
}

double Romb::obwod()
{
    return 4 * a;
}